packages/diffhtml/lib/util/parse.js

```diff
@@ -252,18 +270,25 @@ export default function parse(html, options = {}) {
         continue;
       }
 
-      // If not a doctype, create an element.
-      const newTree = createTree(tagName);
-
+      // Don't call createTree yet, otherwise we won't have access to the
+      // completed element. So create a fake VTree, to build up the object
+      // until we have attributes and child nodes.
+      const newTree = {
```

Comment:
I don't like how this is creating a new object for every tag, instead this should be pulling a VTree from the pool and using that.

Reply:
Unfortunately doing that means that createTree will short-circuit before it gets a chance to run, per:

```js
if (memory.protected.has(input) || memory.allocated.has(input)) {
  entry = input;
}
/*..*/
if (entry) { return entry; }
```

So this probably can't be optimized just yet.